acra/configuring-maintaining/general-configuration/acra-rotate.md

```diff
@@ -0,0 +1,182 @@
+
+* `--dry-run`
+
+  Perform rotation without saving rotated AcraStructs and keys.
+
+### Configuration files
+
+* `--config_file=<filename>`
+
+  Path to YAML configuration file.
+
+* `--dump_config`
+
+  Dump configuration to `configs/acra-addzone.yaml`.
+
+* `--generate_markdown_args_table`
+
+  Generate markdown file with text description of all flags.
+  Output file is `configs/markdown_acra-addzone.md`.
+  Works in pair with `--dump_config`.
+
+### Network
+
+* `--db_connection_string=<db_connection_url>`
+
+  Connection string to DB.
+
+### Storage destination
+
+#### Filesystem
+
+* `--keys_dir=<path>`
+
+  Folder from which keys will be loaded.
+  Default is `.acrakeys`.
+
+* `--file_map_config=<path>`
+
+  Path to file with map of <ZoneId>: <FilePaths> in json format {"zone_id1": ["filepath1", "filepath2"], "zone_id2": ["filepath1", "filepath2"]}.
+
+#### Redis
+
+* `--redis_db_keys=<number>`
+
+  Redis database number to use.
+  Default is `0`.
+  <!-- `acra-server -help` says default is `-1` but in `cmd/redis.go` I see `redisDefaultDB = 0` -->
+  <!-- this var is also used as default value for the flag, where's the truth? -->
+
+* `--redis_host_port=<host:port>`
+
+  Address of Redis database to use as keystore.
+  If not specified, Redis is not used.
+
+* `--redis_password=<password>`
+
+  Password to Redis database.
+
+
+#### HashiCorp Vault
+
+`acra-rotate` can read `ACRA_MASTER_KEY` from HashiCorp Vault instead of environment variable.
+
+* `--vault_connection_api_string=<url>`
+
+  Connection string (like `https://example.com:8200`) for connecting to HashiCorp Vault.
+  If not specified, `ACRA_MASTER_KEY` environment variable will be used.
+
+* `--vault_secrets_path=<kv-path>`
+
+  Path to KV Secrets directory in Vault used to store `ACRA_MASTER_KEY`.
+  Default is `secret/`.
+
+* `--vault_tls_ca_path=<path>`
+
+  Path to CA certificate bundle to use for HashiCorp Vault certificate validation.
+
+  If not specified, use root certificates configured in system.
+
+* `--vault_tls_client_cert=<path>`
+
+  Path to client TLS certificate used to connect to HashiCorp Vault.
+
+  If not specified, don't send client certificate.
+
+* `--vault_tls_client_key=<path>`
+
+  Path to the private key of the client TLS certificate used to connect to HashiCorp Vault.
+
+  If not specified, don't send client certificate.
+
+* `--vault_tls_transport_enable={true|false}`
+
+  Use TLS to encrypt transport with HashiCorp Vault.
+  Default is `false`.
+
+
+## Usage example
+
+Acra encrypts data using storage encryption keys and these keys support rotation process, which can be done in response to a potential leak or compromise.
+
+`acra-rotate` generates new storage keypair, reads data from database via AcraServer, decrypts it and re-encrypts with new keys. In this example, new keys overwrite previous ones, so it's important to rotate all data at the same time.
```

Comment:
are you sure that they overwrite previous, not moving them into archive folder and use for decryption only?

Reply:
Technically speaking, yes it overwrites, but you are right it worth saying that new keys will be moved to `.old` folder for decryption purposes